### PR TITLE
Fix Sparsity Unit Test

### PR DESCRIPTION
**Pull request by Satrat**
SUMMARY:
Fix for failing test detailed in https://app.asana.com/0/1208216504433080/1208216504433083/f. The issue was the quantization modifier was affecting the sparsity, and we weren't checking the mask in the appropriate location. 

TEST PLAN:
Failing nightly test now passes, confirmed locally on janice

### CODE REVIEW

File: src/llmcompressor/modifiers/obcq/utils/sgpt_wrapper.py [hunk 1]
```diff
@@ -83,6 +88,10 @@ def compress(
         final_shape = self.layer.weight.shape
         final_dtype = self.layer.weight.dtype
         W = self.layer.weight.data.clone()
+        args_loc = "quantization_scheme.weights"
+        weight_quant_args = getattr_chain(self.layer, args_loc, None)
+        if weight_quant_args is not None:
+            W = maybe_calibrate_or_quantize(self.layer, W, "weight", weight_quant_args)
```
Comment (dsikka) [inline src/llmcompressor/modifiers/obcq/utils/sgpt_wrapper.py:94]: why do we need this additional call to `maybe_calibrate_or_quantize`?

File: src/llmcompressor/modifiers/obcq/utils/sgpt_wrapper.py [hunk 2]
```diff
@@ -198,6 +209,8 @@ def compress(
         if isinstance(self.layer, transformers.Conv1D):
             W = W.t()
         W = W.reshape(final_shape).to(final_dtype)
+        if weight_quant_args is not None:
+            W = maybe_calibrate_or_quantize(self.layer, W, "weight", weight_quant_args)
```
Comment (dsikka) [inline src/llmcompressor/modifiers/obcq/utils/sgpt_wrapper.py:213]: why do we need this additional call to maybe_calibrate_or_quantize?

File: src/llmcompressor/modifiers/obcq/utils/sgpt_wrapper.py [hunk 3]
```diff
@@ -114,6 +123,14 @@ def compress(
                 torch.tensor(1, dtype=torch.bool),
                 torch.tensor(0, dtype=torch.bool),
             )
+            current_sparsity = mask.sum() / W.numel()
+            print(current_sparsity, sparsity)
```
[the remark this answers is not in the document]
Comment (Satrat) [reply src/llmcompressor/modifiers/obcq/utils/sgpt_wrapper.py]: oops, done!

File: tests/llmcompressor/transformers/obcq/recipes/quant_and_sparse.yaml [hunk 4]
```diff
@@ -13,12 +13,13 @@ test_stage:
           weights:
             num_bits: 8
           targets: ["Linear"]
+    GPTQModifier:
```
[the remark this answers is not in the document]
Comment (Satrat) [reply tests/llmcompressor/transformers/obcq/recipes/quant_and_sparse.yaml:16]: QuantizationModifier and GPTQModifier should be in order, otherwise we end up in a situation were the "naive" minmax quantization is used for the first sparsity calculation, then the more accurate GPTQ quantization is applied afterwards. We want sparsity and quantization to be fully separate operations, not apply sparsity in the middle of the algorithm

File: src/llmcompressor/modifiers/obcq/utils/sgpt_wrapper.py (hunk 2 above)
Comment (Satrat) [reply src/llmcompressor/modifiers/obcq/utils/sgpt_wrapper.py:212]: the sparseGPT algorithm adjusts the weights and they aren't guaranteed to be in the quantized bins, we need to requantize after updating
